apps/studio/src/server/modules/folder/folder.router.ts

```diff
@@ -220,4 +274,111 @@ export const folderRouter = router({
         return pick(result, defaultFolderSelect)
       },
     ),
+
+  createIndexPage: protectedProcedure
+    .input(createIndexPageSchema)
+    .mutation(async ({ ctx, input: { resourceId, siteId } }) => {
+      await validateUserPermissionsForResource({
+        siteId,
+        action: "create",
+        userId: ctx.user.id,
+        resourceId,
+      })
+
+      return await db.transaction().execute(async (tx) => {
+        const { title } = await tx
+          .selectFrom("Resource")
+          .where("Resource.type", "=", ResourceType.Folder)
+          .select("title")
+          .executeTakeFirstOrThrow(
+            () =>
+              new TRPCError({
+                code: "NOT_FOUND",
+                message: "Could not find parent folder to create in",
+              }),
+          )
+
+        const indexPageBlob = await tx
+          .insertInto("Blob")
+          .values({
+            content: jsonb(createIndexPage(title)),
+          })
+          .returning("id")
+          .executeTakeFirstOrThrow()
+
+        const indexPage = await tx
+          .insertInto("Resource")
+          .values({
+            parentId: resourceId,
+            draftBlobId: indexPageBlob.id,
+            title,
+            type: ResourceType.IndexPage,
+            permalink: INDEX_PAGE_PERMALINK,
+            siteId,
+          })
+          .returningAll()
+          .executeTakeFirstOrThrow()
+
+        const by = await tx
+          .selectFrom("User")
+          .where("id", "=", ctx.user.id)
+          .selectAll()
+          .executeTakeFirstOrThrow(
+            () =>
+              new TRPCError({
+                code: "BAD_REQUEST",
+                message: "User could not be found",
+              }),
+          )
+
+        await tx
+          .updateTable("Resource")
+          .where("id", "=", String(indexPage.id))
+          .set({ draftBlobId: indexPageBlob.id })
+          .execute()
+
+        await logResourceEvent(tx, {
+          eventType: "ResourceCreate",
+          delta: { before: null, after: indexPage },
+          by,
+          siteId,
+        })
+
+        return indexPage
+      })
+    }),
+
+  getIndexpage: protectedProcedure
+    .input(getIndexpageSchema)
+    .query(async ({ ctx, input: { resourceId, siteId } }) => {
+      await validateUserPermissionsForResource({
+        siteId,
+        action: "read",
+        userId: ctx.user.id,
+        resourceId,
+      })
+
+      return db.transaction().execute(async (tx) => {
```

Comment:
not necessary to place them in a TX since this theres only DB reads

Reply:
sounds good